src/Util/Asserts.h

```diff
@@ -13,24 +13,40 @@
 #define CONVERT_TO_STRING(X) #X
 #define CONVERT_EXPR_TO_STRING(X) CONVERT_TO_STRING(X)
 
-#define OWSAssert(X) \
-if (!(X)) { \
-NSLog(@"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X)); \
-NSAssert(0, @"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X)); \
-}
-
-// OWSAssert() should be used in Obj-C and Swift methods.
-// OWSCAssert() should be used in free functions.
-#define OWSCAssert(X) \
-if (!(X)) { \
-NSLog(@"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X)); \
-NSCAssert(0, @"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X)); \
-}
+// OWSAssert() and OWSFail() should be used in Obj-C methods.
+// OWSCAssert() and OWSCFail() should be used in free functions.
+
+#define OWSAssert(X)                                                                                                   \
+    if (!(X)) {                                                                                                        \
+        NSLog(@"%s Assertion failed: %s", __PRETTY_FUNCTION__, CONVERT_EXPR_TO_STRING(X));                             \
+        NSAssert(0, @"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X));                                               \
+    }
+
+#define OWSCAssert(X)                                                                                                  \
+    if (!(X)) {                                                                                                        \
+        NSLog(@"%s Assertion failed: %s", __PRETTY_FUNCTION__, CONVERT_EXPR_TO_STRING(X));                             \
+        NSCAssert(0, @"Assertion failed: %s", CONVERT_EXPR_TO_STRING(X));                                              \
+    }
+
+#define OWSFail(X)                                                                                                     \
```

Comment:
Oh and since we're gabbing, I also intend to modify these macros to take NSString-style  format + varargs arguments.  But I haven't done it yet.